FraihmworkHealthAndIntegrityApi.yaml

```diff
@@ -795,6 +795,244 @@ paths:
+        - Start and end times must be provided in the format "HH:MM" and can be used to define precise periods of unavailability within each day.
+      operationId: addUnavailabilityScheduleUsingPOST
+      requestBody:
+        description: The unavailability schedule data for one or more components
+        required: true
+        content:
+          application/json:
+            schema:
+              type: array
+              items:
+                $ref: '#/components/schemas/RecurringSchedule'
+      responses:
+        201:
+          description: Successfully added the unavailability schedule(s)
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Response'
+        400:
+          description: Bad request - Invalid schedule or overlapping schedules.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Response'
+        401:
+          description: Not authenticated
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Response'
+        403:
+          description: Forbidden - client is not authorized to add the schedule
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Response'
+        406:
+          description: Not acceptable - only responds in application/json format
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Response'
+        500:
+          description: Server error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Response'
+      security:
+      - OAuth2: [fraihmwork.component.write, fraihmwork.component.admin]
+
+    get:
+      tags:
+      - Components
+      summary: Get the unavailability schedule for one or more components.
+      description: |
+        Retrieve the unavailability schedule for one or more components.
+        - Provide one or more `componentId` values as query parameters to filter the results.
+        - If no `componentId` is provided, schedules for all components will be retrieved.
+        - Each component has only one time `zone` for all schedules.
+        - The response will return a separate unavailability schedule for each component.
+        - Each day within a component's unavailability schedule will be returned individually, whether it has a time period or not.
+        - If a day has no unavailability scheduled, it will be included with an empty `timePeriods` array.
+        - The order in which days and schedules are returned is not guaranteed and should not be relied upon.
+      operationId: getUnavailabilityScheduleUsingGET
+      parameters:
+      - in: query
```

Comment:
Time zones add a layer of extra complexity on the whole thing!
We should probably allow for an optional parameter to specify the timezone output

Reply:
Something like:

- in: query
        name: timezone
        description: The desired time zone of the output. Should be specified as either a standard time zone id (UTC, GMT, EST, etc.) or as an offset as seen in RFC3339 (+01:00, -02:30, etc.).
        required: false
        schema:
          type: string